pkg/txhandler/txhandler.go

```diff
@@ -27,6 +27,7 @@ import (
 
 type TransactionPersistence interface {
 	persistence.TransactionPersistence
+	RichQuery() persistence.RichQuery
```

Comment:
I don't think this is quite the right way to slice this - will make a counter proposal